tests/mixin-field-test.toit

```diff
@@ -0,0 +1,125 @@
+// Copyright (C) 2023 Toitware ApS.
+// Use of this source code is governed by a Zero-Clause BSD license that can
+// be found in the tests/LICENSE file.
+
+import expect show *
+
+mixin MixA:
+  field-A/int? := null
+  field-A2 := null
```

Comment:
No tests with final fields? Maybe we will get them with the constructor change.

Reply:
also added a test for the type-check of a field.
done.